ObjcVoiceQuickstart/ViewController.m

```diff
@@ -686,4 +709,46 @@ - (void)performAnswerVoiceCallWithUUID:(NSUUID *)uuid
     }
 }
 
+#pragma mark - Ringtone
+
+- (void)playRingtone {
+    NSString *ringtonePath = [[NSBundle mainBundle] pathForResource:@"ringtone" ofType:@"wav"];
+    if ([ringtonePath length] <= 0) {
+        NSLog(@"Can't find sound file");
+        return;
+    }
+    
+    self.ringtonePlayer = [[AVAudioPlayer alloc] initWithContentsOfURL:[NSURL URLWithString:ringtonePath] error:nil];
+    self.ringtonePlayer.delegate = self;
+    self.ringtonePlayer.numberOfLoops = -1;
+    
+    [self play];
+}
+
+- (void)stopRingtone {
+    if (!self.ringtonePlayer.isPlaying) {
+        return;
+    }
+    
+    [self.ringtonePlayer stop];
+    NSError *error = nil;
+    if (![[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord
+                                                error:&error]) {
+        NSLog(@"Failed to reset AVAudioSession category: %@", [error localizedDescription]);
+    }
+}
+
+- (void)play {
+    self.ringtonePlayer.volume = 1.0f;
+    [self.ringtonePlayer play];
+}
+
+- (void)audioPlayerDidFinishPlaying:(AVAudioPlayer *)player successfully:(BOOL)flag {
+    NSError *error = nil;
+    if (![[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord
+                                                error:&error]) {
+        NSLog(@"Unable to reset AVAudioSession category: %@", [error localizedDescription]);
+    }
```

Comment:
what if we don't set the category again? This can cause race condition and result into no audio.

Reply:
👍 Yeah these are the lines copied from the legacy quickstart, which does not use CallKit. In the world with CallKit, we should let CallKit and the audio device handle this. Tested and worked fine without resetting the category. Removing this.